### PR TITLE
UI tool 1072 vitest in mantine

### PR DESCRIPTION
**Pull request by FelixBlaisThon**
### Proposed Changes

https://coveord.atlassian.net/browse/UITOOL-1072

I've made the first step to get Vitest running instead of Jesters. I will do the react package as well, but I would like to first have your feedback on this and apply the same reviews/comments on the react package. I just noticed that I left the jest script command in package.json (I will remove them in an incoming commit) and that I will have to change the README.md in the unit test section. But as for the conversion from jest to vitest, it's ready :)

If green -> Youpi

### Acceptance Criteria

-   [x] The proposed changes are covered by unit tests
-   [ ] The potential breaking changes are clearly identified
-   [ ] [README.md](https://github.com/coveo/plasma/blob/master/README.md) is adjusted to reflect the proposed changes (if relevant)

### CODE REVIEW

File: packages/mantine/__mocks__/monaco-editor.ts [hunk 1]
```diff
@@ -1,6 +1,8 @@
 const editor = {
     create: () => ({
-        dispose: (): void => null,
+        dispose: (): void => {
+            return null;
+        },
```
Comment (GermainBergeron) [inline packages/mantine/__mocks__/monaco-editor.ts:5]:
```suggestion
        dispose: (): void => null,
```

File: packages/mantine/__mocks__/@monaco-editor/react.tsx [hunk 2]
```diff
@@ -17,8 +17,6 @@ const MockedEditor: FunctionComponent<EditorProps> = (props) => {
 
 export default MockedEditor;
 
-export const loader = {
-    config: jest.fn(),
+export const loader: any = {
```
Comment (GermainBergeron) [inline packages/mantine/__mocks__/@monaco-editor/react.tsx:20]: Can you use another type than any?

File: packages/mantine/package.json [hunk 3]
```diff
@@ -68,7 +69,8 @@
         "react-dom": "18.2.0",
         "rimraf": "3.0.2",
         "tslib": "2.4.0",
-        "typescript": "4.9.3"
+        "typescript": "4.9.3",
+        "vitest": "^0.28.3"
```
Comment (GermainBergeron) [inline packages/mantine/package.json:73]:
```suggestion
        "vitest": "0.28.3"
```

File: packages/mantine/src/components/code-editor/CodeEditor.tsx [hunk 4]
```diff
@@ -173,6 +173,7 @@ export const CodeEditor: FunctionComponent<CodeEditorProps> = (props) => {
         </Group>
     );
 
+    // console.log(loader, Editor);
```
Comment (GermainBergeron) [inline packages/mantine/src/components/code-editor/CodeEditor.tsx:176]:
```suggestion
```

File: packages/mantine/src/components/code-editor/__tests__/CodeEditor.spec.tsx [hunk 5]
```diff
@@ -1,24 +1,26 @@
...
-        jest.clearAllMocks();
+        vi.clearAllMocks();
     });
 
     it('renders the monaco editor and a copy to clipboard button', async () => {
-        jest.mock('monaco-editor');
         render(<CodeEditor label="label" description="description" />);
 
-        await waitForElementToBeRemoved(screen.queryByRole('presentation'));
+        await waitForElementToBeRemoved(screen.queryByRole('presentation'), {timeout: 5000});
```
Comment (GermainBergeron) [inline packages/mantine/src/components/code-editor/__tests__/CodeEditor.spec.tsx:19]: We still need that timeout?

File: packages/mantine/src/components/date-range-picker/__tests__/DateRangePickerPopoverCalendar.spec.tsx [hunk 6]
```diff
@@ -67,11 +70,12 @@ describe('DateRangePickerPopoverCalendar', () => {
         await user.click(screen.getAllByRole('button', {name: '14'})[0]);
 
         // hides the calendar when the second date is clicked
-        expect(screen.queryByRole('button', {name: '8'})).not.toBeInTheDocument();
+        expect(screen.queryAllByRole('button', {name: '8'})[0]).not.toBeVisible();
+        expect(screen.queryAllByRole('button', {name: '8'})[1]).not.toBeVisible();
```
Comment (GermainBergeron) [inline packages/mantine/src/components/date-range-picker/__tests__/DateRangePickerPopoverCalendar.spec.tsx:74]: They are in the document?

File: packages/mantine/src/components/table/__tests__/TableDateRangePicker.spec.tsx [hunk 7]
```diff
@@ -11,20 +11,23 @@ const columns: Array<ColumnDef<RowData>> = [columnHelper.accessor('name', {enabl
...
+    const actual = await vi.importActual('@mantine/hooks');
+    return {
+        ...actual,
+        useReduceMotion: () => true,
+    };
+});
 
 describe('Table.DateRangePicker', () => {
-    jest.setTimeout(15000);
+    // vi.setTimeout(15000);
```
Comment (GermainBergeron) [inline packages/mantine/src/components/table/__tests__/TableDateRangePicker.spec.tsx:23]: Remove?

**Review by GermainBergeron — COMMENTED** [review]
Looks good! Just a few minor comments

File: packages/mantine/src/components/date-range-picker/__tests__/DateRangePickerPopoverCalendar.spec.tsx (hunk 6 above)
Comment (FelixBlaisThon) [reply packages/mantine/src/components/date-range-picker/__tests__/DateRangePickerPopoverCalendar.spec.tsx:74]: yes.. Only not visible (as seen from the logTestingPlaygroundURL() method)

File: packages/mantine/src/components/code-editor/__tests__/CodeEditor.spec.tsx (hunk 5 above)
Comment (FelixBlaisThon) [reply packages/mantine/src/components/code-editor/__tests__/CodeEditor.spec.tsx:19]: I will try it out, I just finished finnally fixing all the cypress tests

File: packages/mantine/__mocks__/@monaco-editor/react.tsx (hunk 2 above)
Comment (FelixBlaisThon) [reply packages/mantine/__mocks__/@monaco-editor/react.tsx:20]: from my understanding its a namespace, which does not seem to be usable as a type. Is there something else I can do ?

File: packages/mantine/src/__tests__/VitestSetup.ts [hunk 8]
```diff
@@ -0,0 +1,21 @@
+import '@testing-library/jest-dom';
```
[the remark this answers is not in the document]
Comment (FelixBlaisThon) [reply packages/mantine/src/__tests__/VitestSetup.ts:1]: From a quick check, it extends matchers from expect such as `toHaveValue` or `toHaveAccessibleNames`, but I used something else to extend them from the global expect function from vi
```
import matchers from '@testing-library/jest-dom/matchers';
expect.extend(matchers);
```

[the remark this answers is not in the document]
Comment (FelixBlaisThon) [reply packages/mantine/src/__tests__/VitestSetup.ts:1]: I also found out that we could extend matchers from there but I haven't had a deep dive in it https://vitest.dev/guide/extending-matchers.html

[the remark this answers is not in the document]
Comment (GermainBergeron) [reply packages/mantine/src/__tests__/VitestSetup.ts:1]: Thanks for the info!

File: packages/mantine/__mocks__/@monaco-editor/react.tsx (hunk 2 above)
Comment (GermainBergeron) [reply packages/mantine/__mocks__/@monaco-editor/react.tsx:20]: Can you simply omit the type like it was before? TypeScript should infer it

Comment (FelixBlaisThon) [reply packages/mantine/__mocks__/@monaco-editor/react.tsx:20]: I get the `not portable, type annotation necessary` error from typerscript when I remove the any

**Review by GermainBergeron — APPROVED** [review]

**Comment by dmgauthier** [comment]
This doesn't look to have been super painful. GG !

**Review by dmgauthier — APPROVED** [review]

**Pull request merged**